cups/tls-darwin.c

```diff
@@ -441,16 +444,15 @@ httpCopyCredentials(
 
   if (!(error = SSLCopyPeerTrust(http->tls, &peerTrust)) && peerTrust)
   {
+    DEBUG_printf(("2httpCopyCredentials: Peer provided %d certificates.", (int)SecTrustGetCertificateCount(peerTrust)));
+
     if ((*credentials = cupsArrayNew(NULL, NULL)) != NULL)
     {
-      CFArrayRef secArray = SecTrustCopyCertificateChain(peerTrust);
-      CFIndex i, count = CFArrayGetCount(secArray);
+      count = SecTrustGetCertificateCount(peerTrust);
 
-      DEBUG_printf(("2httpCopyCredentials: Peer provided %ld certificates.", (long)count));
-      
       for (i = 0; i < count; i ++)
       {
-    SecCertificateRef secCert = (SecCertificateRef)CFArrayGetValueAtIndex(secArray, i);
```

Comment:
CFArrayGetValueAtIndex predates SecTrustGetCertificateCount and is actually supported on more OSes than SecTrustGetCertificateCount.